zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -79,8 +79,8 @@ angular.module('zeppelinWebApp')
   var angularObjectRegistry = {};
 
   var editorModes = {
-    'ace/mode/scala': /^%spark/,
-    'ace/mode/python': /^%pyspark/,
+    'ace/mode/python': /^%(spark\.)?pyspark\s*$/,
+    'ace/mode/scala': /^%spark\s*$/,
```

Comment:
I think it's possible to create a new interpreter group /setting so it might not be called "spark"
how about changing it like the sql one:
```
/^%(\w*\.)?pyspark/
/^%(\w*\.)?spark/
```

btw, it looks like the scala one would not work if the user has it as the default and has not put "%spark" in the paragraph